ts/Core/Pointer.ts

```diff
@@ -2118,50 +2172,142 @@ namespace Pointer {
      * */
 
     export interface ChartPositionObject {
+        /**
+         * Left offset of the chart container.
+         */
         left: number;
+        /**
+         * Horizontal scale of the chart container.
+         */
         scaleX: number;
+        /**
+         * Vertical scale of the chart container.
+         */
         scaleY: number;
+        /**
+         * Top offset of the chart container.
+         */
         top: number;
     }
     export interface AxesCoordinatesObject {
+        /**
+         * Positions on the x-axis.
+         */
         xAxis: Array<AxisCoordinateObject>;
+        /**
+         * Positions on the y-axis.
+         */
         yAxis: Array<AxisCoordinateObject>;
     }
     export interface AxisCoordinateObject {
+        /**
+         * Related axis.
+         */
         axis: Axis;
+        /**
+         * Axis value.
+         */
         value: number;
     }
+    /** @internal */
     export interface CoordinatesObject {
+        /**
+         * The X coordinate of the pointer interaction relative to the chart.
+         */
         chartX: number;
+        /**
+         * The Y coordinate of the pointer interaction relative to the chart.
+         */
         chartY: number;
     }
+    /** @internal */
     export interface EventArgsObject {
+        /**
+         * Chart X position of the pointer.
+         */
         chartX?: number;
+        /**
+         * Chart Y position of the pointer.
+         */
         chartY?: number;
+        /**
+         * Filter to apply to the series.
+         */
         filter?: Function;
+        /**
+         * The point currently being hovered.
+         */
         hoverPoint?: Point;
+        /**
+         * Whether the tooltip is shared.
+         */
         shared?: boolean;
     }
+    /** @internal */
     export interface HoverDataObject {
+        /**
+         * The hovered point.
+         */
         hoverPoint?: Point;
+        /**
+         * All hovered points.
+         */
         hoverPoints: Array<Point>;
+        /**
+         * The hovered series.
+         */
         hoverSeries: Series;
     }
+    /**
+     * Axis-specific data of a selection.
+     *
+     * @interface Highcharts.SelectDataObject
+     */
     export interface SelectDataObject {
+        /**
+         * The selected Axis.
+         */
         axis: Axis;
+        /**
+         * The maximum axis value, either automatic or set manually.
+         */
         max: number;
+        /**
+         * The minimum axis value, either automatic or set manually.
+         */
         min: number;
     }
+    /**
+     * Object for select events.
+     * The primary axes are `xAxis[0]` and `yAxis[0]`. Remember the unit of a
+     * datetime axis is milliseconds since 1970-01-01 00:00:00.
+     *
+     * @interface Highcharts.SelectEventObject
```

Comment:
Mismatch between [the public docs](https://api.highcharts.com/class-reference/Highcharts.SelectEventObject) - some props should be internal (all that do not have doclets) while `preventDefault` is missing here.